### ISSUE TITLE
Add additional API for VideoHash

### ISSUE THREAD

**Issue by qarmin** [Farmadupe/vid_dup_finder_lib#1]
Today I finished implementing this library in czkawka and results can be seen here - https://github.com/qarmin/czkawka/pull/460

It was surprisingly easy to do(most of time took preparing GUI)

In app I use custom system to save cache to text files, so I need info about each element from VideoHash.
I found that currenty it is not possible to do it, so I had to create custom fork.

Changes from this fork are visible here(I used rust formatter, so there is a lot diffs from this tool) - https://github.com/qarmin/vid_dup_finder_lib/commit/a4809772aea8f73c9a22da6fb43df50bfdd1b31d.
It would be good to integrate this or similar changes to project, because this will allow me to work with upstream version of this library and get always the newest features. 

Since I load hashes from file, then I need to create `VideoHash` object manually so I added this function
```
        pub fn with_start_data(
            duration: u32,
            src_path: impl AsRef<Path>,
            hash: [u64; HASH_QWORDS],
            num_frames: u32,
        ) -> Self {
            VideoHash {
                hash,
                num_frames,
                src_path: src_path.as_ref().to_path_buf(),
                duration,
            }
        }
```
Also there is no possibility to get `hash` or `num_frames` from VideoHash so I added this functions
```
    /// The hash.
    pub fn hash(&self) -> &[u64; HASH_QWORDS] {
        &self.hash
    }
    pub fn num_frames(&self) -> u32 {
        self.num_frames
    }

```
The last thing is that `const HASH_QWORDS` is not public.
I need to check if number of loaded hash elements is equal to number of needed hash elements, so access to such constant would be really helpful(for now I have hardcoded number 19)

**Comment by Farmadupe** [comment]
Thanks for the kind words, this is my first crate and I spent a lot of time to make the crate easy to integrate. I was worried that nobody would use it after I published to crates.io.

I apologize for the formatting problem, it is because when I separated the library for the application, I forgot to copy my custom rustfmt.toml. I will restore the default soon.

After looking at the proposed change, I think your intention is to serialize/deserialize VideoHash with a manual implementation that accesses all private fields. I am reluctant to make the proposed change because because it would expose additional private data in the public API. It is already possible to serialize/deserialize to any format (binary, csv, json etc) because VideoHash implements the serialize/deserialize traits from Serde.

I looked at the similar_videos.rs file in czkawka and I think it is possible to serialize/deserialize the FileEntry struct using serde. I will prepare a pull request.

If that request cannot be merged, then I will make the proposed change -- It is better to expose some private data so you do not have to maintain the fork.

(PS. I just noticed now that the VideoHash.path field is marked "pub". This is my error in the API, as I did not intend it to be public.)
